### PR TITLE
Add the pattern preview

### PR DESCRIPTION
**Pull request by amieiro**
## Problem

When you see the strings to translate in a pattern, you don't have a live preview to be able to see the strings in context.

## Solution

This PR adds the live preview for the patterns, using the [filter and action added in GlotPress](https://github.com/GlotPress/GlotPress/pull/1665).

image

image

**Responsive preview (under 1280 px)**

image

## Testing Instructions

You need to use [this GlotPress branch](https://github.com/GlotPress/GlotPress/pull/1665) (or the develop branch if this branch has been merged). 

You need to add the `&pattern_preview=true` text in the URL to enable this preview.

To test this PR:
1. Go to the [patterns page](https://translate.wordpress.org/locale/gl/default/patterns/) in any language.
2. Click on the [link of a pattern](https://translate.wordpress.org/projects/patterns/core/gl/default/?filters%5Bterm%5D=https%3A%2F%2Fwordpress.org%2Fpatterns%2Fpattern%2F3-item-listicle%2F&filters%5Bterm_scope%5D=scope_any&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated_or_rejected_or_changesrequested_or_old&filters%5Buser_login%5D=&filter=Apply+Filters&sort%5Bby%5D=priority&sort%5Bhow%5D=desc) to see the live preview.
3. Click in another projects like [WordPress.org](https://translate.wordpress.org/projects/meta/wordpress-org/gl/default/), a [theme](https://translate.wordpress.org/projects/wp-themes/beauty-salon-lite/gl/default/) or a [plugin](https://translate.wordpress.org/projects/wp-plugins/ultimate-member/dev/gl/default/) to be sure that the preview is only showed in the patterns.

### CODE REVIEW

File: wordpress.org/public_html/wp-content/plugins/wporg-gp-virtual-projects/wporg-gp-virtual-projects.php [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+ */
+class WPorg_Virtual_Projects {
+
+	/**
+	 * Adds filters to subprojects and to project status.
+	 */
+	public function __construct() {
+		add_filter( 'gp_subprojects', array( $this, 'add_virtual_projects' ), 10, 1 );
+		add_filter( 'gp_get_project_status', array( $this, 'gp_get_project_status' ), 10, 4 );
+		add_filter( 'gp_translation_table_after', array( $this, 'add_pattern_preview' ), 10, 2 );
```
Comment (akirk) [inline wordpress.org/public_html/wp-content/plugins/wporg-gp-virtual-projects/wporg-gp-virtual-projects.php]:
```suggestion
		add_filter( 'gp_translation_table_classes', array( $this, 'gp_translation_table_classes' ) );
		add_action( 'gp_translation_table_after', array( $this, 'add_pattern_preview' ), 10, 2 );
```

See https://github.com/GlotPress/GlotPress/pull/1665/files#r1279315477

**Comment by akirk** [comment]
After some hiccups with getting the patch applied and realizing I need to append `&pattern_preview=true`, it works!

**Comment by amieiro** [comment]
@akirk can you review this related [GlotPress PR](https://github.com/GlotPress/GlotPress/pull/1665), so we can include it in the next GlotPress release?

**Review by akirk — APPROVED** [review]
https://github.com/GlotPress/GlotPress/pull/1665 is already approved, we can move ahead with this one!

**Comment by amieiro** [comment]
I have released this PR to production and [I have informed to the community](https://make.wordpress.org/polyglots/2023/09/20/previewing-patterns-during-translation/) about this improvement. I close this PR.

**Pull request closed without merging**